core_impl/src/test/java/io/opencensus/implcore/tags/propagation/TagContextDeserializationTest.java

```diff
@@ -71,6 +75,38 @@ public void testDeserializeEmptyByteArrayThrowException() throws Exception {
   }
 
   @Test
+  public void testDeserializeInvalidTagKey() throws Exception {
+    ByteArrayOutputStream os = new ByteArrayOutputStream();
+    os.write(SerializationUtils.VERSION_ID);
+    os.write(SerializationUtils.VALUE_TYPE_STRING);
+
+    // Encode an invalid tag key and a valid tag value:
+    encodeString("\2key", os);
+    encodeString("value", os);
+    final byte[] bytes = os.toByteArray();
+
+    thrown.expect(TagContextParseException.class);
+    thrown.expectMessage("Invalid tag key: \2key");
+    serializer.fromByteArray(bytes);
+  }
+
+  @Test
+  public void testDeserializeInvalidTagValue() throws Exception {
+    ByteArrayOutputStream os = new ByteArrayOutputStream();
+    os.write(SerializationUtils.VERSION_ID);
+    os.write(SerializationUtils.VALUE_TYPE_STRING);
+
+    // Encode a valid tag key and an invalid tag value:
+    encodeString("key", os);
+    encodeString("val\3", os);
+    final byte[] bytes = os.toByteArray();
+
+    thrown.expect(TagContextParseException.class);
+    thrown.expectMessage("Invalid tag value: val\3");
```

Comment:
Imho the most important part of the change is making sure we see the tag name that failed to parse. That helps the user figure out what they screwed up. Can we make the message include the tag name? We have to do things like this in zipkin to reduce support costs.